models/RwdImage.php

```diff
@@ -464,6 +478,108 @@ public function get_set_sources() {
 		return $sources;
 	}
 
+	/**
+	 * Dynamically resize image.
+	 *
+	 * @param int    $attach_id Attachment ID.
+	 * @param array  $meta_data Attachment meta data.
+	 * @param string $key       Image size key.
+	 * @param int    $width     Image width.
+	 * @param int    $height    Image height.
+	 * @param int    $crop      Crop image.
+	 *
+	 * @return array
+	 */
+	public function resize_image( $attach_id, $meta_data, $key, $width, $height, $crop ) {
+		if ( ! empty( $meta_data ) && ! isset( $meta_data['sizes'][ $key ] ) && $meta_data['width'] <= $width ) {
+			// for usual images for lower image sizes we will use max image size for the bigger sizes.
+			$meta_data['sizes'][ $key ] = array(
+				'width'      => $meta_data['width'],
+				'height'     => $meta_data['height'],
+				'file'       => basename( $meta_data['file'] ),
+				'rwd_width'  => $width,
+				'rwd_height' => $height,
+				'crop'       => $crop,
+			);
+			// save to cache.
+			$this->set_attachment_metadata( $attach_id, $meta_data );
+			// update metadata.
+			wp_update_attachment_metadata( $attach_id, $meta_data );
+
+			return $meta_data;
+		}
+		// set RWD size config
+		if ( ! isset( $meta_data['sizes'][ $key ] ) ) {
+			$meta_data['sizes'][ $key ] = array(
+				'rwd_width'  => $width,
+				'rwd_height' => $height,
+				'crop'       => $crop,
+			);
+		} elseif ( ! isset( $meta_data['sizes'][ $key ]['rwd_width'] ) ) {
```

Comment:
if and elseif have to be merged into 1 condition with OR

Reply:
also if this check required only for the condition below - just add this as a part of condition, without init of this dummy array, which you overwrite below 100%